code/modules/mob/living/carbon/human/species_types/golems.dm

```diff
@@ -47,7 +48,11 @@
 
 /datum/species/golem/random/on_species_gain(mob/living/carbon/C, datum/species/old_species)
 	..()
-	var/list/golem_types = typesof(/datum/species/golem) - src.type
+	var/list/golem_types = subtypesof(/datum/species/golem)

```

Comment:
this never changes, make it a static list init at round start

use a subsystem of some appropriate flavour to set it up and remove non valid types

Reply:
Would assigning it in the first `on_species_gain` work as well?

Reply:
sure

Reply:
Please do the cache

Reply:
Sorry, I've been sick and it slipped my mind. I'll get this done when I get a chance.